source/Patches/EndGameSummary.cs

```diff
@@ -0,0 +1,218 @@
+                playerData.Clear();
+            }
+
+            internal class PlayerInfo
+            {
+                public int PlayerId { get; set; }
+                public string PlayerName { get; set; }
+                public Faction Faction { get; set; }
+                public Role CurrentRole { get; set; }
+                public Modifier Modifier { get; set; }
+                public int TasksCompleted { get; set; }
+                public int TasksTotal { get; set; }
+            }
+        }
+
+        public static System.Collections.Generic.List<PlayerControl> allPlayers = new System.Collections.Generic.List<PlayerControl>();
+        public static bool playerInfoUpdateFlag1Updating = false;
+        public static bool playerInfoUpdateFlag2EndOfGame = false;
+
+        public static void UpdatePlayerDataRPCCall()
+        {
+            var writer = AmongUsClient.Instance.StartRpcImmediately(PlayerControl.LocalPlayer.NetId,
+        (byte)CustomRPC.UpdateGamePlayerControlData, Hazel.SendOption.Reliable, -1);
+            AmongUsClient.Instance.FinishRpcImmediately(writer);
+        }
+
+        public static void GatherPlayerData(byte rpccallid)
+        {
+            EndGameSummary.allPlayers.Clear();
+            EndGameSummary.allPlayers = EndGameSummary.GetAllPlayers();
+            List<PlayerControl> allplayersList = EndGameSummary.allPlayers;
+            UpdatePlayerInfo(allplayersList);
+        }
+
+        public static void UpdatePlayerInfo(List<PlayerControl> players)
+        {
+            List<AdditionalTempData.PlayerInfo> pinfolist = new List<AdditionalTempData.PlayerInfo>();
+            pinfolist.Clear();
+
+            foreach (PlayerControl player in players)
+            {
+                int pid = player.PlayerId;
+                string name = player.name;
+                Role role = Role.GetRole(player);
+                Modifier modifier = Modifier.GetModifier(player);
+                Faction faction = role.Faction;
+                int tasksdone = 0;
+                int taskstotal = 0;
+
+                AdditionalTempData.PlayerInfo pinfo = new AdditionalTempData.PlayerInfo();
+                pinfo.PlayerName = name;
+                pinfo.CurrentRole = role;
+                pinfo.Faction = faction;
+                pinfo.Modifier = modifier;
+
+                PluginSingleton<TownOfUs>.Instance.Log.LogMessage($"PlayerName:- " + player.name);
+                foreach (var task in player.myTasks)
+                {
+                    var firstText = task.name;//.Cast<ImportantTextTask>();
+                    PluginSingleton<TownOfUs>.Instance.Log.LogMessage($"TaskName:- " + task.name);
+                    var modname = "junktonull";
+                    var rolename = "junktonull";
+                    if (!IsNull(pinfo, "Mod"))
+                        modname = modifier.Name;
+
+                    if (!IsNull(pinfo, "Role"))
+                        rolename = role.Name;
```

Comment:
Where is this used?

Reply:
without putting in something to check for null, the table does not load, it could be i am having some other technical issues, but the amount of times i was getting a null exception, i put in something to check if those values are null, please feel free to suggest a better way of doing so, my time is limited, so i am probably doing things in inefficient ways.